plugin/src/main/java/org/opensearch/ml/action/stats/MLStatsNodesTransportAction.java

```diff
@@ -127,21 +156,89 @@ MLStatsNodeResponse createMLStatsNodeResponse(MLStatsNodesRequest mlStatsNodesRe
         }
 
         Map<String, MLModelStats> modelStats = new HashMap<>();
-        // return model level stats
         if (mlStatsInput.includeModelStats()) {
-            for (String modelId : mlStats.getAllModels()) {
-                if (mlStatsInput.retrieveStatsForModel(modelId)) {
-                    Map<ActionName, MLActionStats> actionStatsMap = new HashMap<>();
-                    for (Map.Entry<ActionName, MLActionStats> entry : mlStats.getModelStats(modelId).entrySet()) {
-                        if (mlStatsInput.retrieveStatsForAction(entry.getKey())) {
-                            actionStatsMap.put(entry.getKey(), entry.getValue());
+            CountDownLatch latch = new CountDownLatch(1);
+            boolean isSuperAdmin = isSuperAdminUserWrapper(clusterService, client);
+            searchHiddenModels(ActionListener.wrap(hiddenModels -> {
+                for (String modelId : mlStats.getAllModels()) {
+                    if (isSuperAdmin || !hiddenModels.contains(modelId)) {
+                        if (mlStatsInput.retrieveStatsForModel(modelId)) {
+                            Map<ActionName, MLActionStats> actionStatsMap = new HashMap<>();
+                            for (Map.Entry<ActionName, MLActionStats> entry : mlStats.getModelStats(modelId).entrySet()) {
+                                if (mlStatsInput.retrieveStatsForAction(entry.getKey())) {
+                                    actionStatsMap.put(entry.getKey(), entry.getValue());
+                                }
+                            }
+                            modelStats.put(modelId, new MLModelStats(actionStatsMap));
                         }
                     }
-                    modelStats.put(modelId, new MLModelStats(actionStatsMap));
                 }
+                latch.countDown();
+            }, e -> {
+                log.error("Search Hidden model wasn't successful");
+                latch.countDown();
+            }));
+            // Wait for the asynchronous call to complete
+            try {
+                latch.await(120, TimeUnit.SECONDS);
```

Comment:
Do we really need to wait for so long ? Suggest change to a reasonable timeout so the thread not being blocked for a long time and reduce other part.